src/unix/haiku/mod.rs

```diff
@@ -0,0 +1,719 @@
+use dox::mem;
+
+cfg_if! {
```

Comment:
Can this be pushed down to separate modules? Currently we try to keep each module to have at most one invocation of `cfg_if!` for delegating to sub-modules.